api/area.go

```diff
@@ -46,3 +47,80 @@ func (api *API) getArea(w http.ResponseWriter, req *http.Request) {
 	log.Event(ctx, "getArea Handler: Successfully retrieved area", log.INFO, logdata)
 
 }
+
+//GetAreas is a handler that gets all the areas from MongoDB
+func (api *API) getAreas(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	logData := log.Data{}
+
+	offsetParameter := req.URL.Query().Get("offset")
+	limitParameter := req.URL.Query().Get("limit")
+
+	offset := api.defaultOffset
+	limit := api.defaultLimit
+
+	var err error
+
+	if offsetParameter != "" {
+		logData["offset"] = offsetParameter
+		offset, err = utils.ValidatePositiveInt(offsetParameter)
+		if err != nil {
+			log.Event(ctx, "invalid query parameter: offset", log.ERROR, log.Error(err), logData)
+			err = errs.ErrInvalidQueryParameter
+			handleAPIErr(ctx, err, w, nil)
+			return
+		}
+	}
+
+	if limitParameter != "" {
```

Comment:
I'd be tempted to move the `limitParameter` and `limit` checks above the `offsetParameter` check as if the limit exceeds `api.maxLimit`, it returns to the caller regardless of `offsetParameter `.